server/routes/authenticate.tsx

```diff
@@ -1,14 +1,28 @@
 import passport from 'passport';
 import app from 'server/server';
+import { User } from 'server/models';
+import { isDevelopment } from 'utils/environment';
 
-app.get('/auth/zotero', passport.authenticate('zotero'));
+app.get(
+	'/auth/zotero',
+	(req, res, next) => {
+		if (req.user) {
+			const authRedirectHost = req.get('host');
+			User.update({ authRedirectHost }, { where: { id: req.user.id } });
```

Comment:
Since `User.update` returns a promise, we should handle it like

```ts
if (req.user) {
  User.update({ authRedirectHost }, { where: { id: req.user.id } }).then(() => {
    next();
  }).catch(error => {
    next(error);
  });
}
```

Or, with async/await:

```ts
async (req, res, next) => {
  if (req.user) {
    try {
      await User.update(...);
    } catch (error) {
      next(error);
    }
  } else {
    next();
  }
}
```

Which is what our `wrap` utility does!